src/systems/collision.rs

```diff
@@ -0,0 +1,21 @@
+use crate::{Ball, Player, Velocity};
+use bevy::prelude::*;
+
+pub fn collision_system(
+    mut ball_query: Query<(&Transform, &mut Velocity, &Ball)>,
+    player_query: Query<(&Transform, &Player)>,
+) {
+    if let Ok((ball_transform, mut ball_velocity, _)) = ball_query.single_mut() {
+        // TODO: check if the next frame, instead of the current frame, is a collision
+        for (player_transform, _) in player_query.iter() {
+            if ball_transform
+                .translation
+                .distance(player_transform.translation)
+                < 100.0
```

Comment:
Clever solution! When I run the collision detection, the ball collides with the paddle too far away. 

I was wondering if you came across this function from bevy

`bevy::sprite::collide_aabb::collide`
https://docs.rs/bevy/0.4.0/bevy/sprite/collide_aabb/fn.collide.html

It takes the `position` and `size` of objects `a` and `b` and does a side bounding box collision detection.